pkg/cedar/cedarldap/translated_client.go

```diff
@@ -94,3 +95,45 @@ func (c TranslatedClient) FetchUserInfo(ctx context.Context, euaID string) (*mod
 		EuaUserID:  person.UserName,
 	}, nil
 }
+
+// QueryPersons fetches a user's personal details by their common name
+func (c TranslatedClient) QueryPersons(ctx context.Context, commonName string) ([]*models2.UserInfo, error) {
+	if commonName == "" {
+		appcontext.ZLogger(ctx).Error("No commonName specified; skipping request to CEDAR LDAP")
+		return nil, nil
+	}
+
+	params := operations.NewPersonParams()
+	params.CommonName = &commonName
+	resp, err := c.client.Operations.Person(params, c.apiAuthHeader)
+	if err != nil {
+		appcontext.ZLogger(ctx).Error(fmt.Sprintf("Failed to fetch person from CEDAR LDAP with error: %v", err))
+		return nil, &apperrors.ExternalAPIError{
+			Err:       err,
+			Model:     models.Person{},
+			Operation: apperrors.Fetch,
+			Source:    "CEDAR LDAP",
+		}
+	}
+	if resp.Payload == nil {
+		return nil, &apperrors.ExternalAPIError{
+			Err:       errors.New("failed to return person from CEDAR LDAP"),
+			Model:     models.Person{},
+			Operation: apperrors.Fetch,
+			Source:    "CEDAR LDAP",
+		}
+	}
+
+	// Make a slice of UserInfo objects that's sized to the response
+	userInfoList := make([]*models2.UserInfo, len(resp.Payload.Persons))
```

Comment:
Just out of curiosity, any reason for initializing this with the response size instead of creating an empty array and using `append()` inside the for loop?

Reply:
It's honestly probably not worth the more complicated code to read vs. just using `append()`, but my thinking was that it's more performant to make a slice and do `n` assignments, rather than make a slice and do `n` appends, since each append is making a copy of the slice.